zebra-scan/src/service/scan_task/tests/vectors.rs

```diff
@@ -4,25 +4,23 @@ use std::collections::HashMap;
 
 use color_eyre::Report;
 
-use zebra_chain::block::Height;
-
-use crate::service::ScanTask;
+use crate::{service::ScanTask, tests::mock_sapling_viewing_keys};
 
 /// Test that [`ScanTask::process_messages`] adds and removes keys as expected for `RegisterKeys` and `DeleteKeys` command
 #[tokio::test]
 async fn scan_task_processes_messages_correctly() -> Result<(), Report> {
     let (mut mock_scan_task, cmd_receiver) = ScanTask::mock();
     let mut parsed_keys = HashMap::new();
+    let network = Default::default();
 
     // Send some keys to be registered
     let num_keys = 10;
-    mock_scan_task.register_keys(
-        (0..num_keys)
-            .map(|i| (i.to_string(), (vec![], vec![], Height::MIN)))
-            .collect(),
-    )?;
+    let sapling_keys = mock_sapling_viewing_keys(num_keys as u8);
```

Comment:
This is much cleaner, thank you.

Optional/Nitpick:

```suggestion
    let sapling_keys = mock_sapling_viewing_keys(num_keys.try_into().expect("should fit in u8"));
```

Reply:
I thought `as` was a suitable shortcut here since it's not a production code, and `num_keys` is manually set one line above to a low value.